### PR TITLE
MDS-4072: Populate dropdown list project leads

### PR DESCRIPTION
**Pull request by areyeslo**
# Main

- Implement the configuration to assign project leads in party component.

# Other

- Seeded data to add all project leads provided in ticket.
- Added project leads to static data.

# How to test

- Create a user.

Access Project Lead user:
- When selecting the user, you can see Project Lead as a Role and an open ending date .
- Click on Update Contact button at the top right corner.

Deactivate Project Lead role:
- Fill the required data and disable the Project Lead role for the user.
- In history, the user with Project Lead role has an ending date of today.

Reactivate Project Lead role:
- Update again the user to activate the Project Lead role and set the start date to today.
- The user gets the Project Lead role activated again.

Assign Inspector role:
- Update the user to activate the Inspector role and set the start date to today
- The user gets both roles: Project Lead and Inspector

# Notes
https://bcmines.atlassian.net/browse/MDS-4072

### CODE REVIEW

File: services/core-api/app/api/mines/variances/resources/variance.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def put(self, mine_guid, variance_guid):
 
         inspector_party_guid = data.get('inspector_party_guid')
         if inspector_party_guid:
-            inspector = Party.find_by_party_guid(inspector_party_guid)
+            inspector = Party.find_inspector_by_party_guid(inspector_party_guid)
```
Comment (JustinMacaulay) [inline services/core-api/app/api/mines/variances/resources/variance.py]: wondering why this change was needed?

File: services/core-api/app/api/parties/party/models/party.py [hunk 2]
```diff
@@ -144,6 +144,12 @@ def name(cls):
     def find_by_party_guid(cls, party_guid):
         return cls.query.filter_by(party_guid=party_guid, deleted_ind=False).one_or_none()
 
+    @classmethod
```
Comment (JustinMacaulay) [inline services/core-api/app/api/parties/party/models/party.py]: was there a performance concern on the find party by guid?

File: services/core-api/app/api/parties/party_appt/models/party_business_role_appt.py [hunk 3]
```diff
@@ -68,6 +68,16 @@ def get_current_business_appointment(cls, party_guid, party_business_role_code):
                     or_(PartyBusinessRoleAppointment.end_date > today,
                         PartyBusinessRoleAppointment.end_date == None)).one_or_none()
 
+
+    @classmethod
+    def get_current_business_appointments(cls, party_guid, party_business_role_code):
```
Comment (JustinMacaulay) [inline services/core-api/app/api/parties/party_appt/models/party_business_role_appt.py:73]: Is this method being used anywhere I couldn't find a reference to it.

File: services/core-web/src/components/Home.js [hunk 4]
```diff
@@ -82,6 +84,8 @@ export class Home extends Component {
   loadStaticContent = () => {
     this.props.loadBulkStaticContent();
     this.props.fetchInspectors();
+    this.props.fetchProjectLeads();
+    console.log(this.props.fetchProjectLeads);
```
Comment (JustinMacaulay) [inline services/core-web/src/components/Home.js]: is this a forgotten log statement?

**Review by JustinMacaulay — COMMENTED** [review]
Just a few comments

File: services/core-api/app/api/mines/variances/resources/variance.py (hunk 1 above)
Comment (areyeslo) [reply services/core-api/app/api/mines/variances/resources/variance.py]: Even thought the inspector_party_guid was passed, I considered that we needed to validate the party_guid is in fact an Inspector. You are right, FE always will send the inspector_party_guid, so no need to confirm.  Getting back to find_by_party_guid.

File: services/core-api/app/api/parties/party/models/party.py (hunk 2 above)
Comment (areyeslo) [reply services/core-api/app/api/parties/party/models/party.py]: Just to confirm the party is an Inspector. Getting back.

File: services/core-api/app/api/parties/party_appt/models/party_business_role_appt.py (hunk 3 above)
Comment (areyeslo) [reply services/core-api/app/api/parties/party_appt/models/party_business_role_appt.py:73]: it is used in party_resource.py when getting the business_roles:
```
 business_roles = PartyBusinessRoleAppointment.get_current_business_appointments(
                existing_party.party_guid, ("INS", "PRL"))
```

File: services/core-web/src/components/Home.js (hunk 4 above)
Comment (areyeslo) [reply services/core-web/src/components/Home.js]: Thanks Justin

File: services/core-web/common/reducers/partiesReducer.js [hunk 5]
```diff
@@ -63,6 +63,11 @@ export const partiesReducer = (state = initialState, action) => {
         ...state,
         inspectors: action.payload.records,
       };
+    case actionTypes.STORE_PROJECT_LEADS:
+      return {
+        ...state,
+        project_leads: action.payload.records,
+      };
```
Comment (mmumby) [inline services/core-web/common/reducers/partiesReducer.js:70]: This should be camelCase - 'projectLeads', and it should be initiated in the initialState as well

File: services/core-web/common/reducers/partiesReducer.js [hunk 6]
```diff
@@ -84,5 +89,8 @@ export const getLastCreatedParty = (state) => state[PARTIES].lastCreatedParty;
 export const getInspectors = (state) => state[PARTIES].inspectors;
 export const getInspectorsList = (state) =>
   createDropDownList(state[PARTIES].inspectors, "name", "party_guid");
+export const getProjectLeads = (state) => state[PARTIES].project_leads;
+export const getProjectLeadsList = (state) =>
+  createDropDownList(state[PARTIES].project_leads, "name", "party_guid");
```
Comment (mmumby) [inline services/core-web/common/reducers/partiesReducer.js:91]: 'getProjectLeadsList' should be in the selector, but I see that Inspectors is also incorrectly transformed here. We should be importing all redux data into components via selectors, not reducers.

**Review by mmumby — DISMISSED** [review]
Please create tests for all new selectors and actionCreators.

File: services/core-web/common/reducers/partiesReducer.js (hunk 5 above)
Comment (areyeslo) [reply services/core-web/common/reducers/partiesReducer.js:70]: Agreed. Changing to camelCase.

File: services/minespace-web/common/constants/strings.js [hunk 7]
```diff
@@ -76,6 +76,7 @@ export const INCIDENT_DETERMINATION_TYPES = {
 export const INCIDENT_FOLLOWUP_ACTIONS = {
   miu: "MIU",
   inspector: "INS",
+  projectLead: "PRL",
   none: "NO",
```
Comment (mmumby) [inline services/minespace-web/common/constants/strings.js:78]: I don't think this needs to be here - This is for incidents and they do not have a projectLead, only Inspectors.

**Review by mmumby — COMMENTED** [review]
Great work, Happy to Approve if @JustinMacaulay is also good with his comments.

**Review by JustinMacaulay — APPROVED** [review]
Looks good and happy to approve with @mmumby comment

**Pull request merged**